tools/pipeline-owners-extractor/Azure.Sdk.Tools.PipelineOwnersExtractor/Configuration/SecretClientProvider.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+
+using Azure.Core;
+using Azure.Security.KeyVault.Secrets;
+
+namespace Azure.Sdk.Tools.PipelineOwnersExtractor.Configuration
+{
+    public class SecretClientProvider : ISecretClientProvider
+    {
+        private readonly TokenCredential tokenCredential;
+
+        public SecretClientProvider(TokenCredential tokenCredential)
+        {
+            this.tokenCredential = tokenCredential;
+        }
+
+        public SecretClient GetSecretClient(Uri vaultUri)
+        {
+            return new SecretClient(vaultUri, this.tokenCredential);
```

Comment:
What value do we get from a provider interface if the underlying client type is not an interface?

Reply:
In PostConfigureKeyVaultSettings<T>, I need a SecretClient factory that takes a string vaultUri parameter.  Rather than register and take a dependency on Func<string, SecretClient>, I prefer to strongly type the factories.   I could have called it SecretClientFactory, but I don't require new instances on every invocation, so I wend with provider.

The provider itself is an interface that can be mocked in its consumer.  The return type has no interface, but isn't sealed and all of its public methods are virtual, so it can be easily mocked.

Reply:
> How many pipelines/codeowner lookups is this tool going to be doing per run?
We reduce it to a distinct list of aliases and make one lookup per alias.  It's currently processing ~250 aliases in 10-15 seconds.

> In some cases where we parse all pipelines, we have resorted to grabbing the entire github->msft alias mapping via one API call (which takes 10-15s) and caching that, rather than doing individual lookups.
I'd like to see that api info if you have it.

Reply:
Ahh I see, that makes sense, thanks.